packages/website/blog/2024-03-01-parser-options-project-service.md

```diff
@@ -0,0 +1,449 @@
+</table>
+
+### Partial Cold Start
+
+These measurements were taken by recording restarting VS Code on the respective projects with a single file open.
+
+<table>
+<thead>
+<tr>
+<th rowspan="2">API</th>
+<th colspan="2">👶 Tiny</th>
+<th colspan="2">🧒 Small</th>
+<th colspan="2">🧑 Medium</th>
+<th colspan="2">🧓 Large</th>
+</tr>
+<tr>
+<td>Solo</td>
+<td>Projects</td>
+<td>Solo</td>
+<td>Projects</td>
+<td>Solo</td>
+<td>Projects</td>
+<td>Solo</td>
+<td>Projects</td>
+</tr>
+</thead>
+<tbody>
+<tr>
+  <th>Legacy Program</th>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <th>🆕 Project Service</th>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <th>𝚫 Delta</th>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+</tbody>
+</table>
+
+### Warm Start
+
+These measurements were taken by recording VS Code on the respective projects with a single file open and modifying that file.
+
+<table>
+<thead>
+<tr>
+<th rowspan="2">API</th>
+<th colspan="2">👶 Tiny</th>
+<th colspan="2">🧒 Small</th>
+<th colspan="2">🧑 Medium</th>
+<th colspan="2">🧓 Large</th>
+</tr>
+<tr>
+<td>Solo</td>
+<td>Projects</td>
+<td>Solo</td>
+<td>Projects</td>
+<td>Solo</td>
+<td>Projects</td>
+<td>Solo</td>
+<td>Projects</td>
+</tr>
+</thead>
+<tbody>
+<tr>
+  <th>Legacy Program</th>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <th>🆕 Project Service</th>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+<tr>
+  <th>𝚫 Delta</th>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+  <td></td>
+</tr>
+</tbody>
+</table>
+
+## Next Steps
+
+### Giving Feedback
+
+We'd love to hear from you on how this option works for you.
+Does it live up to what we've promised, and/or does it have bugs we haven't fixed yet?
+Please do post in the [Community Feedback: Project Service APIs](https://github.com/typescript-eslint/typescript-eslint/discussions/8030) GitHub Discussions post on how it goes for you.
+
+For support in setting up the new APIs, feel free to ask on [the typescript-eslint Discord](https://discord.gg/FSxKq8Tdyg)'s `#project-service` channel.
+We'd be happy to help you try out `parserOptions.projectService`.
+
+### Long Term Vision
+
+Our hope is that the Project Service API becomes the standard way to work with typed linting over the next few major versions.
+Our intent is to roll it out according to the following rough timeline:
+
+- **v7**: Rename `parserOptions.EXPERIMENTAL_useProjectService` to `parserOptions.projectService`
+- **v8**: Rename `parserOptions.project` to something like `parserOptions.DEPRECATED_legacyProjectProgram` and rename `parserOptions.projectService` to `parserOptions.project`
```

Comment:
I don't entertain well the idea of a new option taking over the old name, unless all v6 configs are valid v8 configs (but in that case why add the `DEPRECATED_` alias?). This makes it harder to identify valid configuration. Outdated 3rd party materials are going to stick forever.

Reply:
Sorry @Josh-Cena I don't follow - what's the issue with this? That resources referring to `parserOptions.projectService` will be out of date?

If so: would preserving it as a deprecated alias for a major version fix that in your opinion?

Reply:
I think you make a good point - and we can just leave this bit out of the blog post as TBD.